tests/e2e.sh

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env bash
+#
+# E2E tests for kpt-functions-catalog.
+
+set -eo pipefail
+
+REPO="$(cd "$(dirname "${BASH_SOURCE[0]}")"/.. >/dev/null 2>&1 && pwd)"
+TAG=${TAG:-dev}
+NODOCKER=${NODOCKER:-}
+HELM_USAGE_SNIPPET="Render chart templates locally"
+
+############################
+# Test framework
+############################
+
+function testcase() {
+  echo "testcase: ${1}"
+  tmp=$(mktemp -d "/tmp/e2e.${1}.XXXXXXXX")
+  cp -r "${REPO}"/example-configs "${tmp}"
+  cp -r "${REPO}"/sh/test-data/* "${tmp}"
+  cd "${tmp}"
+}
+
+function fail() {
+  echo "FAIL: " "$@"
+  exit 1
+}
+
+function assert_contains_string() {
+  content="$(<"$1")"
+  grep -q "$2" "$1" || fail "String $2 not contained in: ${content}"
+}
+
+function assert_dir_exists() {
+  [[ -d $1 ]] || fail "Dir not exist: $1"
+}
+
+############################
+# Docker Tests
+############################
+
+[[ -z ${NODOCKER} ]] || {
+  echo "Skipping docker tests"
+  exit 0
+}
+
+testcase "docker_helm_template_usage_too_few_args"
+docker run -v "$(pwd)/helm":/source gcr.io/kpt-functions/helm-template:"${TAG}" name=too-few-args 2>err.txt || true
+assert_contains_string err.txt "${HELM_USAGE_SNIPPET}"
+
+testcase "docker_helm_template_usage_too_many_args"
+docker run -v "$(pwd)/helm":/source gcr.io/kpt-functions/helm-template:"${TAG}" chart_path=/source/redis name=too-many-args extra=args_provided 2>err.txt || true
+assert_contains_string err.txt "${HELM_USAGE_SNIPPET}"
+
+testcase "docker_helm_template_correct_args"
+docker run -v "$(pwd)/helm":/source gcr.io/kpt-functions/helm-template:"${TAG}" name=correct-args chart_path=/source/redis >out.yaml
```

Comment:
Extract pwd/helm into a variable and reuse

Reply:
It doesn't really make sense extracting the $(pwd) since we'd have to declare a new variable for each test case.  I extracted the common parts of the path to a variable.